File: app/controllers/map_controller.rb
##
# map_controller.rb
#
# university:       University of Applied Sciences Salzburg
# degree course:    MultiMediaTechnology (BSc)
# usage:	        Multimediaprojekt 3 (MMP3)
# authors:          Stephan Griessner
#                   Manuel Mitterer
#                   Franziska Oberhauser

class MapController < ApplicationController
    before_filter :store_last_location
    skip_authorization_check

    # root path
    # GET /
    # GET /tags/:tag
    def index
        @resources = Resource.tagged_with(params[:tag]) if params[:tag]
        @season_tags = Tag.season_tags
        @door_tags = Tag.door_tags
        @level_tags = Tag.level_tags
    end
end

File: app/assets/javascripts/controllers/map_controller.js
/**
 * map_controller.js
 *
 * university:      University of Applied Sciences Salzburg
 * degree course:   MultiMediaTechnology (BSc)
 * usage:           Multimediaprojekt 3 (MMP3)
 * authors:         Stephan Griessner
 *                  Manuel Mitterer
 *                  Franziska Oberhauser
 */

var app = window.app || {};

(function(window, namespace) {

    "use strict";

    /**
     * MapController with mapModel and mapView
     * @param mapModel
     * @param mapView
     * @constructor
     */
    var MapController = function(mapModel, mapView) {

        this._model = mapModel;
        this._view = mapView;

        this._view.locationUpdated.register(function(sender, place) {
            this.updateLocation(place.placeArgs);
        }.bind(this));
    };

    /**
     * notify model about place changes in the view.
     * This happens if the user inserts a new place
     */
    MapController.prototype.updateLocation = function(place) {
        this._model.setLocation(place);
    };

    namespace.MapController = MapController;

})(window, app);